src/cloudflare/internal/vectorize-api.ts

```diff
@@ -16,144 +16,164 @@ enum Operation {
   VECTOR_DELETE = 5,
 }
 
-class VectorizeIndexImpl implements VectorizeIndex {
+/*
+ * The Vectorize beta VectorizeIndex shares the same methods, so to keep things simple, they share one implementation.
+ * The types here are specific to Vectorize GA, but the types here don't actually matter as they are stripped away
+ * and not visible to end users.
+ */
+class VectorizeIndexImpl implements Vectorize {
   public constructor(
     private readonly fetcher: Fetcher,
-    private readonly indexId: string
+    private readonly indexId: string,
+    private readonly indexVersion: string
   ) {}
 
-  public async describe(): Promise<VectorizeIndexDetails> {
-    const res = await this._send(
-      Operation.INDEX_GET,
-      `indexes/${this.indexId}`,
-      {
-        method: "GET",
-      }
-    );
+  public async describe(): Promise<VectorizeIndexInfo> {
+    const endpoint =
+      this.indexVersion === "v2" ? `info` : `binding/indexes/${this.indexId}`;
```

Comment:
For v2, account info and index id are included in the base url of the fetcher so these endpoints just become the final path part.